spec/controllers/admins/users_controller_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe Admins::UsersController, type: :controller do
+  let(:admin) { create(:admin) }
+  let!(:user) { create(:user, &:confirm) }
+  let!(:users) do
+    users = create_list(:user, 20)
+    users.each(&:confirm)
+    users
+  end
+  let!(:pets) do
+    create_list(:pet, 5, user:)
+  end
```

Comment:
destroy の時までリソースを作成する必要はないので index の describe に含めてしまうのが良さそうです。

あと users のところは以下のように書けるかなと思いました。

```suggestion
  let!(:users) do
    create_list(:user, 20, &:confirm)
  end
  let!(:pets) do
    create_list(:pet, 5, user:)
  end
```